addons/mrp/tests/test_bom.py

```diff
@@ -2443,6 +2444,54 @@ def test_archive_operation(self):
         self.assertEqual(bom.bom_line_ids[1].operation_id, ope_2)
         self.assertEqual(bom.byproduct_ids[1].operation_id, ope_2)
 
+    def add_variant_to_product(self, product):
+        attr = self.env['product.attribute'].create({'name': 'Attribute'})
+        self.env['product.attribute.value'].create({
+            'name': 'Variant', 'attribute_id': attr.id, 'sequence': 1
+        })
+        product_template = product.product_tmpl_id
+        attr_line = self.env['product.template.attribute.line'].create({
+            'product_tmpl_id': product_template.id,
+            'attribute_id': attr.id,
+            'value_ids': [Command.set(attr.value_ids.ids)]
+        })
+        return (attr_line)
+
+    def test_onchange_product_id(self):
+        '''
+        When changing variant of the bom, ensure flushing of variant info in bom line.
+        '''
+        attributes = self.add_variant_to_product(self.product_4)
+        self.bom_1.bom_line_ids[0].write({'bom_product_template_attribute_value_ids': [Command.link(attributes.product_template_value_ids[0].id)]})
+        with Form(self.bom_1) as bom:
+            with self.assertLogs(level="WARNING") as warning_logger:
+                bom.product_id = self.product_4.product_tmpl_id.product_variant_ids[0]
+            self.assertEqual(len(warning_logger.output), 1, "Exactly one warning should be logged")
+            self.assertIn(
+                "Changing the product or variant will permanently reset all previously encoded variant-related data.",
+                warning_logger.output[0],
+                "Unexpected warning message"
```

Comment:
```suggestion
```
Useless since it does not provide any information
It's a bit like doing
```py
self.assertEqual(value, 10, "Value should be 10")
```
If the value is not 10, you would have this error message:
```
AssertionError: 1 != 3 : Value should be 10
``` 
We will agree that the error message does not provide anything
You should provide an error message when we need to understand the reason why we are expecting such a value (why this one and not another one)
Most of the time, the test and its assert are clear enough and do not need any details